src/wasm.h

```diff
@@ -1228,9 +1228,12 @@ class RefNull : public SpecificExpression<Expression::RefNullId> {
   void finalize(Type type);
 };
 
-class RefIsNull : public SpecificExpression<Expression::RefIsNullId> {
+class RefIs : public SpecificExpression<Expression::RefIsId> {
 public:
-  RefIsNull(MixedArena& allocator) {}
+  RefIs(MixedArena& allocator) {}
+
+  // RefIs can represent ref.is_null, ref.is_func, ref.is_data, and ref.is_i31.
+  enum What { Null, Func, Data, I31 } what;
```

Comment:
Why not use a typical `...Op` enum for this?

Reply:
Good question. In a way the others reflect the spec, and this reflects an internal detail of Binaryen IR - at least I think that's why it feels slightly better to me. Another is that this can't fit in the global namespace ("Null", "Func", etc.) so it's actually nice to write `RefIs::Null` as in this PR.

But I'm open to other ideas. If we use a normal op, maybe
```cpp
enum RefIsOp
  RefIsNull,
  RefIsFunc,
  RefIsData,
  RefIsI31,
};
```
?

Reply:
That suggested `RefIsOp` enum looks good to me. I do actually like having the public member enums to get the namespacing syntax, though. What would you think of moving all the `...Op` enums to be children of the relevant expression classes as separate change?

Reply:
Ok, sounds good. You're right, it's better to be consistent here.

And yes, let's think about moving all the Op enums separately. That could be good.